recon/cli.py

```
@@ -0,0 +1,179 @@
+    # Merge config defaults
+    args = {**resolve_defaults(config, module), **args}
+
+    http = HTTPClient(cache_ttl=cache_ttl, use_cache=not no_cache)
+    # Inject http client if module expects it
+    if "http" not in args:
+        args["http"] = http
+
+    # Validate metasploit-like required options
+    ok, missing = mod.validate(args)
+    if not ok:
+        raise typer.BadParameter(f"Missing required options for {module}: {', '.join(missing)}")
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)